node/actors/executor/src/attestation.rs

```diff
@@ -207,50 +41,23 @@ impl AttestationStatusRunner {
     /// Poll the client forever in a loop or until canceled.
     async fn poll_forever(&mut self, ctx: &ctx::Ctx) -> ctx::Result<()> {
         loop {
-            self.poll_until_some(ctx).await?;
-            ctx.sleep(self.poll_interval).await?;
-        }
-    }
-
-    /// Poll the client until some data is returned and write it into the status.
-    async fn poll_until_some(&mut self, ctx: &ctx::Ctx) -> ctx::Result<()> {
-        loop {
-            match self.client.attestation_status(ctx).await {
-                Ok(Some((genesis, next_batch_to_attest))) => {
-                    self.status.update(genesis, next_batch_to_attest).await?;
-                    return Ok(());
-                }
-                Ok(None) => {
-                    tracing::info!("waiting for attestation status...")
-                }
-                Err(error) => {
-                    tracing::error!(
-                        ?error,
-                        "failed to poll attestation status, retrying later..."
-                    )
+            match self.client.config(ctx).await {
+                Ok(Some(config)) => {
+                    self.state
+                        .update_config(config)
+                        .await
+                        .context("update_config")?;
                 }
+                Ok(None) => tracing::debug!("waiting for attestation config..."),
+                Err(error) => tracing::error!(
+                    ?error,
+                    "failed to poll attestation config, retrying later..."
+                ),
+            }
+            if let Err(ctx::Canceled) = ctx.sleep(self.poll_interval).await {
```

Comment:
Isn't [this](https://github.com/matter-labs/era-consensus/blob/3bb3e190189dfdcf32e231b06087cec733dee676/node/actors/executor/src/attestation.rs#L36) enough?